tileserver/__init__.py

```diff
@@ -273,12 +298,19 @@ def handle_request(self, request):
                 tile_data = reformat_selected_layers(
                     tile_data_all, self.layer_config.all_layers, coord, format)
 
-                # note that we want to store the data too, as this means that
-                # future requests can be serviced directly from the store.
+                # note that we want to store the formatted data too,
+                # as this means that future requests can be serviced
+                # directly from the store.
                 if self.store and coord.zoom <= 20:
                     self.io_pool.apply_async(
                         async_store, (self.store, tile_data, coord, format))
 
+            # since we're a request for all, enqueue the coordinate to
+            # ensure other formats get processed
+            if self.sqs_queue and coord.zoom <= 20:
```

Comment:
If the (JSON) tile data was in the store, then we'd have returned on [line 257](https://github.com/mapzen/tileserver/pull/14/files#diff-6f8345e485a005306db95fdc97de8d19R257), wouldn't we? In which case, once we store the JSON and enqueue a re-render, we shouldn't end up in this branch more than once per tile (modulo it concurrently happening on several tileservers, and us potentially returning `None` from tile fetching because there was an error rather than no tile).

So, while a request for, say, `buildings` would end up at the tileserver again, it should be able to generate it from the JSON we stored last time rather than come through this branch again? Which my very circuitous way of trying to say I think it's safe to enqueue anything that gets added to the TOI.